src/Validator.js

```diff
@@ -255,15 +255,41 @@ class Validator {
 				if (typeof config === 'object') {
 					// Merge editorconfig values into the correct settings names:
 					for (key in config) {
-						if (typeof MAPPINGS[key] === 'string') {
+						if (typeof MAPPINGS[key] === 'object') {
+							// Disable the setting for "unset" values from editorconfig.
+							// See: Issue #47
+							if (config[key] === 'unset') {
+								this._settings[MAPPINGS[key].name] = false;
+								continue;
+							}
+
+							// Disable the setting for invalid types from editorconfig.
+							// See: Issue #47
+							if (!MAPPINGS[key].types.includes(typeof config[key])) {
+								this._settings[MAPPINGS[key].name] = false;
+								continue;
+							}
+
+							// Disable the setting for invalid string values from editorconfig
+							// by testing against a regular expression.
+							// See: Issue #47
+							if (
+								typeof config[key] === 'string' &&
+								MAPPINGS[key].regexp instanceof RegExp &&
+								!MAPPINGS[key].regexp.test(config[key])
```

Comment:
I think I will merge the three `if`, like that:
```javascript
// Handle "unset" special value given by editorconfig file, see issue #47
// Consider invalid value as no check, see issue #47
if (
    config[key] === 'unset' || 
    !MAPPINGS[key].types.includes(typeof config[key]) ||
    (
        typeof config[key] === 'string' &&
        MAPPINGS[key].regexp instanceof RegExp &&
	!MAPPINGS[key].regexp.test(config[key])
    )
) {
```

Reply:
Hmm, I see what you mean that all three conditions result in `continue;`, but they do it for different reasons. Combining them into a single condition as you suggest, makes it harder to understand. We could add comments in between to separate the three logical groups, but that would almost bring us back to the same outcome.

```js
// Handle parsed values given by editorconfig file, see issue #47 
if (
  // Handle "unset" special value:
  config[key] === 'unset' ||

  // Ignore unexpected or invalid types:
  !MAPPINGS[key].types.includes(typeof config[key]) ||

  // Skip not allowed string values:
  (
    typeof config[key] === 'string' && 
    MAPPINGS[key].regexp instanceof RegExp &&
    !MAPPINGS[key].regexp.test(config[key])
  )
) {
  continue;
}
```

I do like the new comments a bit more 😄, but I still prefer the three separate `if (...) { ... }` blocks, as they’re easier to read. In the end, this is just a stylistic choice, right?

Reply:
Yes just a stylistic choice.

If it was my code, I will only comment the `unset` part as it is special that we do this check. For the rest of the code, reading it makes the situation pretty clear that we are checking that config match with some regexp expectation.

In that matter, `unset` could be part of the regexp to simplify the code.

My approach is to comment only unexpected code. I see the comment as a justification for explananing why not all cases are considered or why some special treatment are done. Filtering correct value is not really a special treatment 😄 
To give you another example of good/bad comment (in my opinion):
```javascript
// Assign the default value to the variable => BAD
const myVar = 'someValue'

if (type === 'myType') {
    // When type = myType, array contains only one element => GOOD
    const myVar = myArray[0]
    ...
}
```

Reply:
I hope I am far away from `// Assign the default value to the variable` 😆. Your suggestion works for me, and I’ve just made a slight adjustment to the comment above:

```js
// Handle "unset" special value given by editorconfig file
// and consider not to parse invalid types and value.
// See: Issue #47
if (
    config[key] === 'unset' || 
    !MAPPINGS[key].types.includes(typeof config[key]) ||
    (
        typeof config[key] === 'string' &&
        MAPPINGS[key].regexp instanceof RegExp &&
	!MAPPINGS[key].regexp.test(config[key])
    )
) {
```